dataset/include/scipp/dataset/util.h

```diff
@@ -0,0 +1,20 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// Copyright (c) 2020 Scipp contributors (https://github.com/scipp)
+/// @file
+/// @author Matthew Andrew
+#pragma once
+#include <scipp/dataset/dataset.h>
+#include <scipp/variable/variable.h>
+
+namespace scipp {
+SCIPP_DATASET_EXPORT scipp::index size_of(const VariableConstView &view);
+
+SCIPP_DATASET_EXPORT scipp::index size_of(const Variable &view);
```

Comment:
Why is this overload required?

Reply:
This is due to how the classes access the underlying `VariableConcept`, In the view this is view.underlying().data(), whereas in the actual variable it is var.data()

Reply:
True, but for a variable implicitly converted to a view `underlying()` will just give the variable, so it should give the same without the code duplication..

Reply:
Good point, is not needed.